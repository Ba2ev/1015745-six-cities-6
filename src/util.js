import dayjs from 'dayjs';
import {offers} from './mock';

export const getDateTimeFormat = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

export const getLocalizedDate = (date) => {
  return dayjs(date).format(`YYYY-MM`);
};

export const getCityOffers = (city) => {
  return offers.filter(({city: {name}}) => name === city);
};
