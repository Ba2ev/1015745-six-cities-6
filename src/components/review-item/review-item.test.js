import React from 'react';
import {render} from '@testing-library/react';
import ReviewItem from './review-item';
import {getLocalizedDate} from '../../util/util';

const review = {
  "id": 1,
  "user": {
    "id": 18,
    "isPro": true,
    "name": `Sophie`,
    "avatarUrl": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`
  },
  "rating": 2,
  "comment": `Beautiful space, fantastic location and atmosphere, really a wonderful place to spend a few days. Will be back.`,
  "date": `2021-03-07T08:04:28.647Z`
};

it(`ReviewItem should render correctly`, () => {
  const {queryByText} = render(<ReviewItem review={review}/>);

  const textTime = queryByText(getLocalizedDate(review.date));
  const textName = queryByText(review.user.name);
  const textRatingValue = queryByText(review.rating);

  expect(textTime).toBeInTheDocument();
  expect(textName).toBeInTheDocument();
  expect(textRatingValue).not.toBeInTheDocument();
});


it(`ReviewItem should render snapshot correctly`, () => {
  const {container} = render(<ReviewItem review={review}/>);

  expect(container).toMatchSnapshot();
});
