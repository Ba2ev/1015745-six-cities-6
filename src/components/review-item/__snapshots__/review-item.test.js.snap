// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewItem should render snapshot correctly 1`] = `
<div>
  <li
    class="reviews__item"
  >
    <div
      class="reviews__user user"
    >
      <div
        class="reviews__avatar-wrapper  user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          class="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg"
          width="54"
        />
      </div>
      <span
        class="reviews__user-name"
      >
        Sophie
      </span>
    </div>
    <div
      class="reviews__info"
    >
      <div
        class="reviews__rating rating"
      >
        <div
          class="reviews__stars rating__stars"
        >
          <span
            style="width: 40%;"
          />
          <span
            class="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        class="reviews__text"
      >
        Beautiful space, fantastic location and atmosphere, really a wonderful place to spend a few days. Will be back.
      </p>
      <time
        class="reviews__time"
        datetime="2021-03-07"
      >
        March 2021
      </time>
    </div>
  </li>
</div>
`;
